onnxruntime/core/optimizer/gather_fusion.cc

```diff
@@ -175,4 +175,118 @@ Status GatherToSplitFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
 
   return Status::OK();
 }
+
+/*
+Fuse Range->Gather to Slice.
+*/
```

Comment:
Can you add a note here explaining that this fusion is primarily helpful for gradient computation for context.